config/config_test.go

```diff
@@ -0,0 +1,50 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package config
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/otel/metric/noop"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func TestNewSDK(t *testing.T) {
+	tests := []struct {
+		name           string
+		cfg            []ConfigurationOption
+		tracerProvider any
+		meterProvider  any
+		err            error
+	}{
+		{
+			name:           "no-configuration",
+			tracerProvider: trace.NewNoopTracerProvider(),
+			meterProvider:  noop.NewMeterProvider(),
+		},
+		{
+			name: "with-configuration",
+			cfg: []ConfigurationOption{
+				WithContext(context.Background()),
+				WithOpenTelemetryConfiguration(OpenTelemetryConfiguration{
+					TracerProvider: &TracerProvider{},
```

Comment:
Side note:

The OTel Go SDK misses "introspection" API 😞 
How do you plan to test more complex setup (e.g. with that an exporter)?
Should we add "accessors" (getters) to the Go SDK?  Do you plan to use `reflect`? Other ideas?

Reference: https://blog.ploeh.dk/2013/04/04/structural-inspection/

Reply:
I was going to ask the same thing. Another option would be to test providers with full end-to-end setup, not super easy but at least then we'll know for sure the components work as intended. This could get somewhat tricky for testing all the permutations of options though

Reply:
Personally, I think it would be good to add  "introspection" API  (add getters and export some types) in OTel Go SDK (if the SIG agrees that it would be beneficial).
Before it is done, the probably most efficient way to avoid rewriting all tests in future would be to use `reflect` to access private fields and types. Later the `reflect` could be changed to use the newly exported API (getters and types).

We can also discuss it further during SIG meeting.

Reply:
I created https://github.com/open-telemetry/opentelemetry-go/issues/4606. Resolving this conversation.